src/cxx.cc

```diff
@@ -100,6 +108,18 @@ String::String(const char *s, std::size_t len) {
              len);
 }
 
+String::String(const char16_t *s) {
+  assert(s != nullptr);
+  initString(this, s, std::char_traits<char16_t>::length(s));
+}
+
+String::String(const char16_t *s, std::size_t len) {
+  assert(s != nullptr || len == 0);
+  initString(this,
+             s == nullptr && len == 0 ? reinterpret_cast<const char16_t *>(1) : s,
```

Comment:
This is UB because 1 is not a legal pointer value for the \&\[u16\] unsafely constructed by slice::from_raw_parts on the Rust side. See the documentation in https://doc.rust-lang.org/1.54.0/std/slice/fn.from_raw_parts.html#safety about alignment.

Reply:
Interesting, thanks for pointing that out. I changed to it 2, guessing that is correct alignment for a slice with 2 byte long elements?